src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/NotifyClaimCallbackHandler.java

```diff
@@ -190,6 +190,11 @@ private CallbackResponse submitClaim(CallbackParams callbackParams) {
             claimDetailsNotificationDeadline = getDeadline(claimNotificationDate);
         }
 
+        if (claimDetailsNotificationDeadline.isAfter(caseData.getClaimNotificationDeadline())
```

Comment:
I can see that the code just above this is setting the claimDetailsNotificationDeadline based on feature flag.

` if (featureToggleService.isCertificateOfServiceEnabled() && areAnyRespondentsLitigantInPerson(caseData)) {
            claimDetailsNotificationDeadline = getDeadline(getServiceDate(caseData));
        } else {
            claimDetailsNotificationDeadline = getDeadline(claimNotificationDate);
        }`

Your code is going to overwrite the deadline date no matter what feature flag's value is. Is that what is required?

Reply:
This logic is for calculated date weather its calculated by COS or else block . In case that date is crossing 4 months deadline then it should be overwrite and not cross 4 month deadline